cabal-install/src/Distribution/Client/CmdHaddock.hs

```diff
@@ -141,7 +141,8 @@ haddockAction flags@NixStyleFlags {..} targetStrings globalFlags = do
     runProjectPostBuildPhase verbosity baseCtx buildCtx' buildOutcomes
   where
     verbosity = fromFlagOrDefault normal (configVerbosity configFlags)
-    cliConfig = commandLineFlagsToProjectConfig globalFlags flags mempty -- ClientInstallFlags, not needed here
+    flags' = flags { installFlags = installFlags { installDocumentation = Flag True } }
```

Comment:
Can it be that this unconditional setting leads to `--disable-doc` being ignored?

Reply:
Yes, I think you're right.